File: Dynamic Programing/Fibonacci/C++/fibonacci_in_space_optimized.cpp
#include <iostream>
using namespace std;
int fib(int n)
{
    int a=0,  b=1, c;
    if(n == 0) return a;
    for(int i=2; i<=n; i++){
        c = a+b;
        a = b;
        b = c;
    }
    return c;
}
int main()
{
    int n;
    while(1){
        cin >> n;
        if(n <= 0) break;
        cout << "Fibonacci of " << n << "th number is " << fib(n) << endl;
    }
    return 0;
}

File: Dynamic Programing/Fibonacci/PHP/fibonacci_in_space_optimized.php
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Fibonacci in space optimized</title>
</head>
<body>
    <form action="" method="post">
        <input type="number" name="number">
        <button>=</button>
    </form>

    <?php
    function fib($n)
    {
        $a=0;
        $b=1;
        $c;
        for($i=2; $i<=$n; $i++){
            $c = $a + $b;
            $a = $b;
            $b = $c;
        }
        return $b;
    }

    if(isset($_POST["number"])){
        $n = $_POST["number"];
        echo fib($n);
    }

    ?>
</body>
</html>